### PR TITLE
Update to Dunfell

### PR DESCRIPTION
**Pull request by jakogut**
Changes were tested with the generic-aarch64 and generic-amd64 device types from https://github.com/balena-os/balena-generic/pull/15 in QEMU. 

Machines booted with EFI; networking, SSH and mDNS work. Machines connected to the dashboard, could be joined to applications, and successfully downloaded and ran those applications.

Not tested:
- HUP
- Any platform specific drivers

### CODE REVIEW

**Comment by klutchell** [comment]
VersionBot is mad cause you need to replace `meta-resin` here:

https://github.com/balena-os/balena-generic/blob/4536093000c79d96f291ac4a907e5090f7b3ed8c/versionist.conf.js#L26

There are also a couple documentation updates here:

https://github.com/balena-os/balena-generic/blob/4536093000c79d96f291ac4a907e5090f7b3ed8c/layers/meta-resin-generic/conf/samples/local.conf.sample#L5
https://github.com/balena-os/balena-generic/blob/4536093000c79d96f291ac4a907e5090f7b3ed8c/README.md#L17

**Comment by floion** [comment]
@resin-jenkins test this please

**Comment by floion** [comment]
@jakogut versionbot is complaining so:

> Command failed: versionist
error: pathspec 'layers/meta-resin' did not match any file(s) known to git.
Could not find meta-resin submodule

**Comment by klutchell** [comment]
> VersionBot is mad cause you need to replace `meta-resin` here:
> 
> https://github.com/balena-os/balena-generic/blob/4536093000c79d96f291ac4a907e5090f7b3ed8c/versionist.conf.js#L26
>

**Comment by jakogut** [comment]
> > VersionBot is mad cause you need to replace `meta-resin` here:
> > https://github.com/balena-os/balena-generic/blob/4536093000c79d96f291ac4a907e5090f7b3ed8c/versionist.conf.js#L26

Yep, looks like I missed that.

File: versionist.conf.js [hunk 1]
```diff
@@ -22,15 +22,15 @@ const compareExtendedSemver = (a, b) => {
   return a.localeCompare(b)
```
Comment (alexgg) [inline versionist.conf.js:22]: @nazrhom this `versionist.conf.js` is no longer used and can be removed, right?

File: layers/meta-resin-generic/conf/samples/bblayers.conf.sample [hunk 2]
```diff
@@ -13,7 +13,7 @@ BBLAYERS ?= " \
     ${TOPDIR}/../layers/meta-openembedded/meta-networking \
     ${TOPDIR}/../layers/meta-openembedded/meta-python \
     ${TOPDIR}/../layers/meta-rust \
-    ${TOPDIR}/../layers/meta-resin/meta-resin-common \
-    ${TOPDIR}/../layers/meta-resin/meta-resin-pyro \
     ${TOPDIR}/../layers/meta-resin-generic \
+    ${TOPDIR}/../layers/meta-balena/meta-balena-common \
+    ${TOPDIR}/../layers/meta-balena/meta-resin-pyro \
```
Comment (alexgg) [inline layers/meta-resin-generic/conf/samples/bblayers.conf.sample]: Why meta-resin-pyro and not Dunfell?

**Review by alexgg — COMMENTED** [review]
@floion @jakogut Browsing this repo I stumbled across the concept of "dependant devices" which now live in the `dependent-devices` folder. What are these?

File: layers/meta-resin-generic/conf/samples/bblayers.conf.sample (hunk 2 above)
Comment (jakogut) [reply layers/meta-resin-generic/conf/samples/bblayers.conf.sample]: @alexgg This commit renamed this file along with several others without modifications, then updated them to dunfell in a later commit.

Comment (alexgg) [reply layers/meta-resin-generic/conf/samples/bblayers.conf.sample]: thanks @jakogut I missed that

**Review by alexgg — COMMENTED** [review]
@jakogut could you please add a few lines about how this changes were tested?

**Comment by jakogut** [comment]
> @jakogut could you please add a few lines about how this changes were tested?

Done. It wasn't extensive testing, but this repo sticks pretty close to meta-balena. It boots, networking works, it can be joined to an application and apps run as expected.

**Review by alexgg — APPROVED** [review]
lgtm

I opened PRs to add versionbot support and remove the dependant devices folder too.

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Comment by alexgg** [comment]
@resin-jenkins ok to test

**Comment by alexgg** [comment]
@jakogut I sorted out the jenkins build jobs but the builds fail with:
```
ERROR: make-mod-scripts-1.0-r0 do_configure: oe_runmake failed
ERROR: make-mod-scripts-1.0-r0 do_configure: Execution of '/yocto/resin-board/build/tmp/work/generic_armv7ahf-poky-linux-gnueabi/make-mod-scripts/1.0-r0/temp/run.do_configure.32418' failed with exit code 1:
make: *** /yocto/resin-board/build/tmp/work-shared/generic-armv7ahf/kernel-source: No such file or directory.  Stop.
WARNING: exit code 1 from a shell command.
```
Could you try to build locally?

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Comment by alexgg** [comment]
@resin-jenkins retest this please

**Pull request merged**